pkg/controller/utils/capability.go

```diff
@@ -67,6 +66,26 @@ type CapabilityComponentDefinition struct {
 	CapabilityBaseDefinition
 }
 
+// NewCapabilityComponentDef will create a CapabilityComponentDefinition
+func NewCapabilityComponentDef(componentDefinition *v1beta1.ComponentDefinition) CapabilityComponentDefinition {
+	var def CapabilityComponentDefinition
+	def.Name = componentDefinition.Name
+	if componentDefinition.Spec.Workload.Definition == (commontypes.WorkloadGVK{}) && componentDefinition.Spec.Workload.Type != "" {
+		def.WorkloadType = util.ReferWorkload
+		def.WorkloadDefName = componentDefinition.Spec.Workload.Type
+	}
+	if componentDefinition.Spec.Schematic != nil && componentDefinition.Spec.Schematic.HELM != nil {
+		def.WorkloadType = util.HELMDef
+		def.Helm = componentDefinition.Spec.Schematic.HELM
+	}
+	if componentDefinition.Spec.Schematic != nil && componentDefinition.Spec.Schematic.KUBE != nil {
+		def.WorkloadType = util.KubeDef
+		def.Kube = componentDefinition.Spec.Schematic.KUBE
+	}
+	def.ComponentDefinition = *componentDefinition
```

Comment:
`def.ComponentDefinition = *componentDefinition.DeepCopy()` is safer since comdDef is passed from external as a pointer.

Reply:
good suggestion, I will fix it